t/multi/open_multi.c

```diff
@@ -0,0 +1,191 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+
+#include "t/lib/tap.h"
+#include "t/lib/testutil.h"
+
+#include "mpi.h"
+#include "unifyfs.h"
+
+#define all_ok(condition, comm, ...) \
+    do { \
+        int input = (int) condition; \
+        int output; \
+        MPI_Allreduce(&input, &output, 1, MPI_INT, MPI_LAND, comm); \
+        int rank; \
+        MPI_Comm_rank(comm, &rank); \
+        if (rank == 0) { \
+            ok_at_loc(__FILE__, __LINE__, output, __VA_ARGS__, NULL); \
+        } \
+    } while (0)
```

Comment:
@CamStan , we can eventually move this kind of macro to an mpitap.h file.  I haven't tested this just yet, but it at least compiles.

Reply:
Nice @adammoody! I'll grab this and test it out.